x-pack/plugins/fleet/server/services/spaces/agent_policy.ts

```diff
@@ -166,4 +169,55 @@ export async function updateAgentPolicySpaces({
     ignore_unavailable: true,
     refresh: true,
   });
+
+  const agentIndexExists = await esClient.indices.exists({
+    index: AGENTS_INDEX,
+  });
+
+  // Update agent actions
+  if (agentIndexExists) {
+    const pitId = await openPointInTime(esClient);
+
+    try {
+      let hasMore = true;
+      let searchAfter: SortResults | undefined;
+      while (hasMore) {
+        const { agents } = await getAgentsByKuery(esClient, newSpaceSoClient, {
+          kuery: `policy_id:"${agentPolicyId}"`,
+          showInactive: true,
+          perPage: UPDATE_AGENT_BATCH_SIZE,
+          pitId,
+          searchAfter,
+        });
+
+        if (agents.length === 0) {
+          hasMore = false;
+          break;
+        }
+
+        const lastAgent = agents[agents.length - 1];
+        searchAfter = lastAgent.sort;
+
+        await esClient.updateByQuery({
+          index: AGENT_ACTIONS_INDEX,
```

Comment:
we don't store spaces in `.fleet-actions-results`, right? so that doesn't have to be updated